epub-reader/src/lib/highlight-manager.ts

```
@@ -0,0 +1,1727 @@
+    const commonFailureReasons = Array.from(errorCounts.entries())
+      .sort(([,a], [,b]) => b - a)
+      .slice(0, 3)
+      .map(([type]) => type);
+    
+    // Generate recommendations
+    const recommendations: string[] = [];
+    if (successRate < 80) {
+      recommendations.push('Consider improving text preprocessing or DOM structure');
+    }
+    if (errorCounts.get(HighlightErrorType.CROSS_NODE_TEXT) || 0 > 0) {
+      recommendations.push('Text spans multiple DOM elements - cross-node search is being used');
```

Biome (2.1.2)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)